pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/internal/TreeExportCommand.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cli.commands.internal;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+import org.slf4j.LoggerFactory;
+
+import net.sourceforge.pmd.cli.commands.mixins.internal.EncodingMixin;
+import net.sourceforge.pmd.cli.commands.typesupport.internal.PmdLanguageTypeSupport;
+import net.sourceforge.pmd.cli.internal.ExecutionResult;
+import net.sourceforge.pmd.internal.LogMessages;
+import net.sourceforge.pmd.lang.Language;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+import net.sourceforge.pmd.properties.PropertySource;
+import net.sourceforge.pmd.util.StringUtil;
+import net.sourceforge.pmd.util.log.internal.SimpleMessageReporter;
+import net.sourceforge.pmd.util.treeexport.TreeExportConfiguration;
+import net.sourceforge.pmd.util.treeexport.TreeExporter;
+import net.sourceforge.pmd.util.treeexport.TreeRendererDescriptor;
+import net.sourceforge.pmd.util.treeexport.TreeRenderers;
+
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Mixin;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+
+@Command(name = "ast-dump", description = "Experimental: dumps the AST of parsing source code")
```

Comment:
@oowekyala should I have this command hidden from the base `pmd -h`? you would still be able to use it, ask for `pmd ast-dump -h`, but it wouldn't be listed otherwise.